zboxcore/sdk/copyworker_test.go

```diff
@@ -0,0 +1,437 @@
+				return
+			}
+			require.NoErrorf(err, "expected no error but got %v", err)
+		})
+	}
+}
+
+func TestCopyRequest_copyBlobberObject(t *testing.T) {
+	const (
+		mockAllocationTxId = "mock transaction id"
+		mockClientId       = "mock client id"
+		mockClientKey      = "mock client key"
+		mockRemoteFilePath = "mock/remote/file/path"
+		mockDestPath       = "mock/dest/path"
+		mockAllocationId   = "mock allocation id"
+		mockType           = "f"
+	)
+
+	var mockClient = mocks.HttpClient{}
+	zboxutil.Client = &mockClient
+
+	client := zclient.GetClient()
+	client.Wallet = &zcncrypto.Wallet{
+		ClientID:  mockClientId,
+		ClientKey: mockClientKey,
+	}
+
+	tests := []struct {
+		name     string
+		setup    func(t *testing.T, name string)
+		wantErr  bool
+		errMsg   string
+		wantFunc func(require *require.Assertions, req *CopyRequest)
+	}{
+		{
+			name:    "Test_Error_New_HTTP_Failed_By_Containing_" + string([]byte{0x7f, 0, 0}),
+			setup:   func(t *testing.T, testName string) {},
+			wantErr: true,
+			errMsg:  `parse "Test_Error_New_HTTP_Failed_By_Containing_\u007f\x00\x00": net/url: invalid control character in URL`,
+		},
+		{
+			name: "Test_Error_Get_Object_Tree_From_Blobber_Failed",
+			setup: func(t *testing.T, testName string) {
+				mockClient.On("Do", mock.MatchedBy(func(req *http.Request) bool {
+					return strings.HasPrefix(req.URL.Path, testName)
+				})).Return(&http.Response{
+					StatusCode: http.StatusBadRequest,
+					Body:       ioutil.NopCloser(bytes.NewReader([]byte(""))),
+				}, nil)
+			},
+			wantErr: true,
+			errMsg:  "Object tree error response: Status: 400 -  ",
+		},
+		{
+			name: "Test_Copy_Blobber_Object_Failed",
+			setup: func(t *testing.T, testName string) {
+				mockClient.On("Do", mock.MatchedBy(func(req *http.Request) bool {
+					return req.Method == "GET" &&
+						strings.HasPrefix(req.URL.Path, testName)
+				})).Return(&http.Response{
+					StatusCode: http.StatusOK,
+					Body: func() io.ReadCloser {
+						jsonFR, err := json.Marshal(fileref.ReferencePath{
+							Meta: map[string]interface{}{
+								"type": mockType,
+							},
+						})
+						require.NoError(t, err)
+						return ioutil.NopCloser(bytes.NewReader([]byte(jsonFR)))
+					}(),
+				}, nil)
+				mockClient.On("Do", mock.MatchedBy(func(req *http.Request) bool {
+					return req.Method == "POST" &&
+						strings.HasPrefix(req.URL.Path, testName)
+				})).Return(&http.Response{
+					StatusCode: http.StatusBadRequest,
+					Body:       ioutil.NopCloser(bytes.NewReader([]byte(""))),
+				}, nil)
+			},
+			wantFunc: func(require *require.Assertions, req *CopyRequest) {
+				require.NotNil(req)
+				require.Equal(uint32(0), req.copyMask)
+				require.Equal(float32(0), req.consensus)
+			},
+		},
+		{
+			name: "Test_Copy_Blobber_Object_Success",
+			setup: func(t *testing.T, testName string) {
+				mockClient.On("Do", mock.MatchedBy(func(req *http.Request) bool {
+					return strings.HasPrefix(req.URL.Path, testName)
```

Comment:
Test the `connection_id`, `path` and `dest` fields set in `copyBlobberObject` are in teh http.Request.